charmhub-track-closer.py

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env python3
+
+# Helper script to close tracks; shows the revision number for the track charm
+# indicating which track is going to be closed.
+# Works with sections (openstack,ceph,ovn,misc) and individual charms, and on
+# specific tracks.  Please see help for the command for more details.
+
+import argparse
+import logging
+from pathlib import Path
+from typing import List, Optional, NamedTuple
+import requests
+import subprocess
+import sys
+
+
+SCRIPT_DIR = Path(__file__).parent.resolve()
+sys.path.append(str(SCRIPT_DIR.parent))
+CHARMS_DIR = SCRIPT_DIR / 'charms'
+
+from lib.lp_builder import get_charms, Charm
+from lib.channel_map import decode_channel_map_to_risks, RISKS
+
+# from https://api.snapcraft.io/docs/charms.html
+CHARMHUB_BASE = "https://api.charmhub.io/v2/charms"
+INFO_URL = CHARMHUB_BASE + "/info/{charm}?fields=channel-map"
+
+
+logger = logging.getLogger(__name__)
+
+
+class Release(NamedTuple):
+    charmhub: str
+    track: str
+    edge: Optional[int]
+    beta: Optional[int]
+    candidate: Optional[int]
+    stable: Optional[int]
+
+class NoRelease(NamedTuple):
+    charmhub: str
+    issue: str
+
+
+def close_track(
+    charms: List[Charm],
+    track: str,
+    risk: str,
+    confirmed: bool = False,
+    ignore_errors: bool = False,
+) -> None:
+    """Close a track depending on the arguments passed.
+
+    This uses the track/risk for a list of charms to close a particular track.
```

Comment:
let's merge this functionality, but we must definitively migrate it to charmhub-lp-tools.